app/controllers/service_controller.rb

```diff
@@ -31,7 +31,12 @@ def generic_object_custom_buttons
     display_options = {}
     ids = @lastaction == 'generic_object' ? @sb[:rec_id] : 'LIST'
     display_options[:display] = @display
-    display_options[:display_id] = params[:id]
+    display_options[:display_id] = if params[:miq_grid_checks]
+                                     params[:id]
+                                   else
+                                     # Determine related service ID for selected GO
+                                     @breadcrumbs.last[:url][/\/\d*\?/][1..-2]
+                                   end
```

Comment:
We should not determine ID from breadcrumbs :-(

Reply:
Breadcrumbs are a presentation "detail". The should not be used to determine what's an ID of an object we are going to work with.

Such code is guaranteed to cause regression. Not now. But once someone tries to actually fix this.